aeppl/opt.py

```diff
@@ -178,8 +158,84 @@ def naive_bcast_rv_lift(fgraph, node):
     return [bcasted_node.outputs[1]]
 
 
-logprob_canonicalize = SequenceDB()
+class ScanRV(RandomVariable):
+    """A placeholder used to specify a log-likelihood for a scan sub-graph."""
+
+    @classmethod
+    def create_node(cls, node, scan_rvs):
+        out_var = node.default_output()
+
+        scanrv_op = ScanRV(
+            f"{out_var.name if out_var.name else ''}-scan_rv",
+            out_var.ndim,
+            [_rvs.out.ndim for _rvs in scan_rvs],
+            out_var.dtype,
+            inplace=False,
+        )
+        scanrv_op.num_indices = len(scan_rvs)
+
+        new_node = scanrv_op.make_node(None, node.out.shape, None, *scan_rvs)
+
+        return new_node
+
+    def _infer_shape(self, size, dist_params, param_shapes=None):
+        return size
+
+    def rng_fn(self, rng, *args, **kwargs):
+        raise NotImplementedError()
+
+
+@local_optimizer([Scan])
+def scan_replace(fgraph, node):
+
+    node_outs = node.op.outputs
+
+    if not node_outs:
+        return
+
+    inner_rvs = []
+
+    # GWalk the inner fgraphs for any random variables
+    for node_out in node_outs:
+        out_fgraph = FunctionGraph(
+            outputs=[node_out],
+            clone=True,
+            copy_orphans=False,
+            copy_inputs=False,
+            features=[ShapeFeature()],
+        )
```

Comment:
Is there a way we can get the Scan's inner graph as `FunctionGraph` so we can walk that instead of each output node.

Reply:
Are you referring to the output nodes of the inner-graph?

Reply:
Yeah like making an entire temporary fgraph just to walk it through to find RV's seems a little bit redundant. Is there a better way to detect the RVs.

Reply:
`FunctionGraph`s are simple containers that only hold a few `list`s (and sometimes a `dict`), so don't worry too much about creating and discarding them.  The same goes for `ScanArgs`.